game-core/src/main/java/games/strategy/triplea/delegate/BattleTracker.java

```diff
@@ -948,22 +957,17 @@ public IBattle getPendingBattle(final GUID guid) {
     if (guid == null) {
       return null;
     }
-    for (final IBattle battle : pendingBattles) {
-      if (guid.equals(battle.getBattleId())) {
-        return battle;
-      }
-    }
-    return null;
+
+    return pendingBattles.stream()
+        .filter(b -> b.getBattleId().equals(guid))
```

Comment:
Why are the parameters swapped now? `guid` is guaranteed to be non-null at this point, so `guid.equals` will never throw an NPE.

Reply:
Not a strong reason, it's a bit more canonical in java to have the constant on right hand side, eg: `a == 2`

Double checking, looks like all elements of `pendingBattles` should be non-null and all have a battle id.